src/core/classes/factory.ts

```diff
@@ -15,7 +20,12 @@ export class ModuleFactory extends ContractWrapper<TWFactory> {
     storage: IStorage,
     options?: SDKOptions,
   ) {
-    super(network, TW_FACTORY_ADDRESS, TWFactory__factory.abi, options);
+    super(
+      network,
+      options?.thirdwebModuleFactory as string,
```

Comment:
This will not always be passed, therefore we need to fetch it from the `CONTRACT_ADDRESSES` map